training/Deepspeed-Domino/domino/tensor_parallel/random.py

```diff
@@ -0,0 +1,241 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+# This file is adapted from random.py in Megatron-LM
+
+import contextlib
+
+import torch
+from torch import _C
+from torch.cuda import _lazy_call
+from torch.cuda import device as device_ctx_manager
+from torch.utils.checkpoint import detach_variable
+
+from domino.parallel_state import get_tensor_model_parallel_rank
+from domino.utils import safely_set_viewless_tensor_data
+from .utils import gather_split_1d_tensor, split_tensor_into_1d_equal_chunks
+
+# Default name for the model parallel rng tracker.
+_MODEL_PARALLEL_RNG_TRACKER_NAME = 'model-parallel-rng'
+
+
+def _set_cuda_rng_state(new_state, device=-1):
```

Comment:
are we using cuda RNG? I remember it cannot be used together with cudagraph, but can be used together if cudagraph not enabled.

Reply:
we are using it. it cannot be used together with cudagraph.